cmake/DLAF_AddTest.cmake

```diff
@@ -67,8 +88,19 @@ function(DLAF_addTest test_target_name)
 
   target_link_libraries(${test_target_name} PRIVATE DLAF)
   target_link_libraries(${test_target_name} PRIVATE DLAF_test)
-  if (DLAF_AT_USE_GTEST_MAIN)
-    target_link_libraries(${test_target_name} PRIVATE gtest_main)
+
+  if (DLAF_AT_USE_MAIN)
+    if (DLAF_AT_USE_MAIN STREQUAL PLAIN)
+      target_link_libraries(${test_target_name} PRIVATE gtest_main)
+    elseif (DLAF_AT_USE_MAIN STREQUAL HPX)
+      target_link_libraries(${test_target_name} PRIVATE DLAF_gtest_hpx_main)
+    elseif (DLAF_AT_USE_MAIN STREQUAL MPI)
+      target_link_libraries(${test_target_name} PRIVATE DLAF_gtest_mpi_main)
+    elseif (DLAF_AT_USE_MAIN STREQUAL MPIHPX)
+      message(FATAL_ERROR "USE_MAIN=${DLAF_AT_USE_MAIN} is not yet supported")
```

Comment:
@rasolca, I have implemented the external main for MPI but not for the combo MPIHPX.